### DISCUSSION TITLE
I made a module!  Two-factor-Authentication add-on

### DISCUSSION THREAD

**Discussion by SinisterSpatula** [DSJAS/DSJAS#17] [Show and tell]
I'm very happy with how it came out.  

Video of it here: https://youtu.be/CUJJY8LZlS4

and it's on my github here: https://github.com/SinisterSpatula/DSJAS-two_factor_auth

Let me know how you like it, and if you have any questions.  It's my first adventure into HTML/CSS/JS so I'm no expert but man I'm proud of it :)

**Comment by ejv2** [comment]
Amazing! I will download a copy and give it a try as soon as I can.

**Comment by ejv2** [comment]
Right, I've tested this out and it's brilliant! I particularly like the loading effects and transitions between different menus in the process.

Now, would you be interested in having this merged into mainline DSJAS to be shipped with the site? If so, copy the module into your fork of DSJAS (consider making a new branch) and putting it in the installed modules folder. Then, we can put together some documentation together (see [this](https://github.com/DSJAS/DSJAS/blob/master/docs/builtins/modules/An%20especially%20frustrating%20login%20process.md) as an example of the format). Of course, your name and everything will remain on it and you will be placed in the third party credits file.

Of course, only if you're interested. I'm sure this would make a great addition to the core.

**Reply by SinisterSpatula** [reply]
Awesome!  I'm honestly not that great with github yet and afraid I'd mess something up.  If you wouldn't mind to toss it into the core for me?  Credit isn't necessary, but don't mind if you put it either.  For docs, I have some info here: https://github.com/SinisterSpatula/DSJAS-two_factor_auth/blob/main/README.md

You have my permission to adapt or modify or include or use it.  I personally would like to see the phone number be pulled from the username/accounts table, and if no phone number exists, just show a fully masked number maybe?  But feel free to change it however you like/need.

**Reply by ejv2** [reply]
Of course, I'm completely fine with doing that. But, it would be nice for GitHub to list you as a contributor. If you're absolutely sure, I'll get working on merging it in and writing up some documentation.

**Reply by SinisterSpatula** [reply]
I'll try to do a pull request today.  Let's see if I can. :)
